e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -70,8 +70,21 @@ private void assertThatAccessStructuredWasFetched(String conversationId, Documen
         assertThat(accessStructured.get("taskId")).isNotNull();
     }
 
-    private void assertThatAccessDocumentWasFetched(String documentId, List<Document> accessDocuments) {
-        Document document = accessDocuments.get(0);
+    private Document theDocumentTaskUpdatesTheRecord(String conversationId) {
+        var gpcAccessDocument = (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT);
+        if(gpcAccessDocument != null) {
```

Comment:
Could we reduce the number of indents here? For example my moving some code lines to different methods?

Reply:
fixed